File: 0452-minimum-number-of-arrows-to-burst-balloons/0452-minimum-number-of-arrows-to-burst-balloons.cpp
class Solution {
public:
    static bool mycomp(vector<int> &a, vector<int> &b){
        if(a[1]!=b[1]) return a[1]<b[1];
        else return a[0]<b[0];
    }
    int findMinArrowShots(vector<vector<int>>& points) {
        sort(points.begin(), points.end(), mycomp);
        int val=points[0][1], ans=1;
        for(int i=1; i<points.size(); i++){
            if(points[i][0]> val){
                ans++;
                val= points[i][1];
            }
        }
        return ans;
    }
};

File: 0452-minimum-number-of-arrows-to-burst-balloons/0452-minimum-number-of-arrows-to-burst-balloons.java
class Solution {
    public int findMinArrowShots(int[][] points) {
        Arrays.sort(points, (a, b) -> Integer.compare(a[1], b[1]));
        int lastPoint = points[0][1], ans=0;
        for(int i=1; i<points.length; i++){
            if(lastPoint < points[i][0]){
                ans++;
                lastPoint= points[i][1];
            }
        }
        return ans+1;
    }
}